sympy/functions/elementary/integers.py

```diff
@@ -587,6 +573,42 @@ def _value_one_or_more(self, other):
             if other.is_integer and other.is_positive:
                 return S.true
 
+    def _eval_as_leading_term(self, x, logx=None, cdir=0):
+        from sympy.calculus.accumulationbounds import AccumBounds
+        arg = self.args[0]
+        arg0 = arg.subs(x, 0)
+        r = self.subs(x, 0)
+
+        if arg0.is_finite:
+            if r.is_zero:
+                ndir = arg.dir(x, cdir=cdir)
+                return S.One if ndir.is_negative else arg
```

Comment:
arg could be sin(x).

Reply:
Ohh okay , do you mean that rather than `arg` ,we should do something with `arg.as_leading_term` ? Are the following answers correct ?
```
>>> frac(sin(x)).as_leading_term(x, cdir=1)
sin(x)
>>> frac(sin(x)).as_leading_term(x, cdir=-1)
1
>>> frac(sin(x**2)).as_leading_term(x, cdir=1)
sin(x**2)
>>> frac(sin(x**2)).as_leading_term(x, cdir=-1)
sin(x**2)
```

Reply:
We should take only the leading term of arg instead of all of it.

Reply:
Actually it won't be `arg.as_leading_term(x)`, that we would be using . We would be using `(arg - arg0).as_leading_term(x)` . This is because consider `frac(-2*x + 1)` . Now if `ndir` is `nonnegative`, if we use arg's (-2*x + 1) leading term we get `1`. **Which is obviously incorrect .**

The graph is something like this 
<img width="396" alt="image" src="https://user-images.githubusercontent.com/87052487/189028920-68de533a-11ee-44ed-b917-9bfecd216794.png">

Hence we want `-2*x` as our answer. This can be explained through the following
```
>>> frac(-2*x + 1).rewrite(floor)
-2*x - floor(-2*x)
>>> frac(-2*x).rewrite(floor)
-2*x - floor(-2*x)
>>> _.as_leading_term(x)
-2*x
```

Reply:
Some other cases where this logic plays a role 
```
>>> frac(-2*x + 1).as_leading_term(x)
-2*x
>>> frac(2*x + 1).as_leading_term(x)
2*x
>>> frac(sin(x) + 5).as_leading_term(x)
x
>>> frac(sin(x) + 5).as_leading_term(x, cdir = -1)
1
```

Reply:
> We would be using (arg - arg0).as_leading_term(x) . 

Yes, that's right.